tests/functional/workloads/cnv/test_vm_snapshot_cloning_ops.py

```diff
@@ -75,58 +75,179 @@ def test_vm_clone(
+            vm_obj.stop()
+
+            # Taking Snapshot of PVC
+            snap_obj = snapshot_factory(pvc_obj)
+
+            # Restore the snapshot
+            res_snap_obj = snapshot_restore_factory(
+                snapshot_obj=snap_obj,
+                storageclass=vm_obj.sc_name,
+                volume_mode=snap_obj.parent_volume_mode,
+                access_mode=vm_obj.pvc_access_mode,
+                status=constants.STATUS_BOUND,
+                timeout=300,
+            )
+
+            # Create new VM using the restored PVC
+            res_vm_obj = cnv_workload(
+                volume_interface=constants.VM_VOLUME_PVC,
+                storageclass=vm_obj.sc_name,
+                source_url=constants.CNV_FEDORA_SOURCE,
+                existing_pvc_obj=res_snap_obj,
+                namespace=vm_obj.namespace,
+            )
+
+            # Expand PVC if `pvc_expand_after_restore` is True
+            if pvc_expand_after_restore:
+                new_size = 50
+                try:
+                    res_snap_obj.resize_pvc(new_size=new_size, verify=True)
+                    assert res_vm_obj.get_vm_pvc_obj().size == new_size, (
+                        f"Failed: VM PVC Expansion of {res_vm_obj.name} after "
+                        f"snapshot restore"
+                    )
+
+                    # Get rootdisk name
+                    disk = (
+                        vm_obj.vmi_obj.get()
+                        .get("status")
+                        .get("volumeStatus")[1]["target"]
+                    )
+                    devicename = f"/dev/{disk}"
+
+                    result = vm_obj.run_ssh_cmd(
+                        command=f"lsblk -d -n -o SIZE {devicename}"
+                    ).strip()
+
+                    if result == f"{new_size}G":
+                        log.info("expanded PVC size is showing on vm")
+                    else:
+                        raise ValueError(
+                            "Expanded PVC size is not showing on VM. "
+                            "Please verify the disk rescan and filesystem resize."
+                        )
+                except ValueError as e:
+                    log.error(
+                        f"Error for VM {vm_obj}: {e}. Continuing with the next VM."
+                    )
+                    failed_vms.append(vm_obj.name)
+                    continue
+
+            # Validate data integrity of file written before taking snapshot
+            res_csum = cal_md5sum_vm(vm_obj=res_vm_obj, file_path=file_paths[0])
+            assert (
+                source_csum == res_csum
+            ), f"Failed: MD5 comparison between source {vm_obj.name} and restored {res_vm_obj.name} VMs"
+
+            # Write new file to VM
+            run_dd_io(vm_obj=res_vm_obj, file_path=file_paths[1], verify=True)
+            res_vm_obj.stop()
+        if failed_vms:
+            assert False, f"Test case failed for VMs: {', '.join(failed_vms)}"
```

Comment:
Can we just storing the VM names? Should we also store the VM configuration details? These details can help when asserting and understanding what configuration the VM had when it failed

Reply:
Sure. we can add configuration as well.

Reply:
Added configuration as well.